torch/onnx/_globals.py

```diff
@@ -0,0 +1,31 @@
+"""Flags that control the ONNX exporter."""
+
+from typing import Optional
+
+import torch.onnx
+from torch.onnx import _constants
+
+
+class _InternalFlags:
+    """Flags used internally by ONNX exporter."""
+
+    def __init__(self):
+        self._export_onnx_opset_version = _constants.onnx_default_opset
+        self.operator_export_type: Optional[torch.onnx.OperatorExportTypes] = None
+        self.training_mode = None
+        self.onnx_shape_inference: bool = False
+
+    @property
+    def export_onnx_opset_version(self):
+        return self._export_onnx_opset_version
+
+    @export_onnx_opset_version.setter
+    def export_onnx_opset_version(self, value: int):
+        supported_versions = [_constants.onnx_main_opset]
+        supported_versions.extend(_constants.onnx_stable_opsets)
+        if value not in supported_versions:
+            raise ValueError(f"Unsupported ONNX opset version: {value}")
+        self._export_onnx_opset_version = value
+
+
+_FLAGS = _InternalFlags()
```

Comment:
Can we rename it to GLOBALS too?
Rationale:
The name globals seems to better capture what these are. Flags is less specific, I think. A flag can be anything that controls behavior, whereas this specifically contains global variables.

Also having a leading underscore on the object and on the module seems redundant.
Two options:
1. Keep this here, but rename to remove the leading underscore on the object, so it becomes `from torch.onnx._globals import GLOBALS`.
2.  Move this to `__init__.py` so users would write `from torch.onnx import _GLOBALS`

Up to you which of those you choose.

Reply:
Done. I went with (1) because (2) may complicate the `__init__` and make dependency resolution trickier.